src/main/java/com/google/api/core/ApiFutures.java

```diff
@@ -104,6 +104,26 @@ public void onSuccess(V v) {
     return new ListenableFutureToApiFuture<V>(catchingFuture);
   }
 
+  public static <V, X extends Throwable> ApiFuture<V> catchingAsync(
```

Comment:
Overal looks good to me, but can we mark this as @Beta API just in case?

Reply:
Done